sklearn/cluster/hierarchical.py

```diff
@@ -713,7 +713,7 @@ def fit(self, X, y=None):
         -------
         self
         """
-        X = check_array(X)
+        X = check_array(X, ensure_min_samples=2)
```

Comment:
While I am not opposed to this change, why is this needed? How is it related to the purpose of this PR? Is this something specific in hierarchical clustering? If so we should probably have an inline comment here to motivate this non default value.

If this is because the default value for `n_clusters` is `2` then I think we should have:

```python
X = check_array(X, ensure_min_samples=n_clusters)
```

instead. That would be more explicit.